src/clr-angular/forms/combobox/combobox.ts

```diff
@@ -46,69 +72,228 @@ export function comboboxDomAdapterFactory(platformId: Object) {
+  }
+
+  @HostListener('keyup', ['$event'])
+  onKeyUp(event: KeyboardEvent) {
+    // if BACKSPACE in multiselect mode, delete the last pill if text is empty
+    if (event.keyCode === BACKSPACE && this.multiSelect && this.lastInputString.length === 0) {
+      this.optionSelectionService.selectionModel.pop();
+    }
+
+    // if input text has changed since last time, fire a change event so application can react to it
+    const currentText = this.textbox.nativeElement.value;
+    if (currentText !== this.lastInputString) {
+      this.clrInputChange.emit(currentText);
+      this.optionSelectionService.currentInput = currentText;
+      this.lastInputString = currentText;
     }
   }
 
-  closeMenuOnTabPress(event: KeyboardEvent) {
-    if (event && event.keyCode === TAB) {
-      this.toggleService.open = false;
+  @Output('clrInputChange') public clrInputChange: EventEmitter<string> = new EventEmitter<string>(false);
+
+  @Output('clrOpenChange') public clrOpenChange: Observable<boolean> = this.toggleService.openChange;
+
+  private initializeSubscriptions(): void {
+    this.subscriptions.push(
+      this.optionSelectionService.selectionChanged.subscribe((newSelection: SelectionModel<T>) => {
+        if (this.textbox) {
+          this.writeValue(newSelection.model);
+          // clear out any user-typed value upon selection
+          if (this.textbox) {
+            this.textbox.nativeElement.value = '';
+          }
+          this.showCurrentSingleSelection = true;
+          // for the purposes of validation
+          if (this.onChangeCallback) {
+            this.onChangeCallback(this.optionSelectionService.selectionModel.model);
+          }
+        }
+      })
+    );
+
+    this.subscriptions.push(
+      this.toggleService.openChange.subscribe(open => {
+        if (!open) {
+          if (this.textbox) {
+            this.textbox.nativeElement.value = '';
+          }
+          this.showCurrentSingleSelection = true;
+        }
+      })
+    );
+
+    this.subscriptions.push(
+      this.toggleService.popoverAligned.subscribe(popoverNode => {
+        const popover: HTMLElement = popoverNode as HTMLElement;
+        // Update position if popover hides the label
+        if (popover.getBoundingClientRect().top < this.el.nativeElement.getBoundingClientRect().top) {
+          this.renderer.setStyle(popoverNode, 'top', `${popover.offsetTop + this.offsetService.labelOffset}px`);
+        }
+      })
+    );
+  }
+
+  // ControlValueAccessor implementation methods
+  writeValue(value: T | T[]): void {
+    this.optionSelectionService.setSelectionValue(value);
+  }
+
+  registerOnChange(onChange: any): void {
+    this.onChangeCallback = onChange;
+  }
+
+  private onTouchedCallback = () => {
+    if (this.control) {
+      this.control.control.markAsTouched();
     }
+  };
+
+  registerOnTouched(onTouched: any): void {
+    this.onTouchedCallback = onTouched;
   }
 
+  setDisabledState(isDisabled: boolean): void {}
```

Comment:
I'm not sure why this method is here. Its not called and doesn't seem to be part of the WrappedFormCOntrol extensaion.

Reply:
It's part of the `ControlValueAccessor` interface that we implement, so that we can pretend to be a natural form element.